src/index.js

```diff
@@ -0,0 +1,212 @@
+/**  
+ * translate status code to response message
+ * @type {{[key:number]:string}}
+ */ 
+const response = {
+  400: 'bad request',
+  401: 'unauthorized',
+  403: 'forbidden',
+  404: 'not found',
+  405: 'method not allowed',
+  409: 'conflict',
+  418: 'i am a teapot',
+  420: 'enhance your calm',
+  422: 'unprocessable entity',
+  500: 'internal server error',
+}
+
+/**
+ * Generic HTTP errors have JSON message and status
+ * @class
+ * @extends {Error}
+ */
+class HttpError extends Error {
+  /**
+   * @param {number} statusCode 
+   * @param {string} message 
+   * @param {string | boolean} externalMessage 
```

Comment:
All of these `externalMessage` properties are optional - they need to be in `[]` -
```suggestion
   * @param {string | boolean} [externalMessage]
```

<img width="568" alt="image" src="https://user-images.githubusercontent.com/23663629/217099199-493095e5-1837-46fc-a97c-6babcf49a2e5.png">